invenio_rdm_records/services/config.py

```diff
@@ -231,6 +239,11 @@ class RDMSearchOptions(SearchOptions, SearchOptionsMixin):
         MetricsParam,
     ]
 
+    query_parser_cls = QueryParser.factory(
```

Comment:
for reviewers: I find this very confusing to have to define the query_parser_cls both in `RDM_SEARCH` config variable and here. Does anyone know the reason for this config being spread in many places?